File: 3-Patron/python/CreadorMapaJungla.py
import CreadorDeMapa
import Mazmorra
import Tesoro
import Textura
import Mapa


class CreadorMapaJungla(CreadorDeMapa):
    @property
    def mapa(self):
        return self.__mapa

    def crearMapa(self):
        mazmorra = Mazmorra()
        tesoro = Tesoro(100, 10000)
        textura = Textura("nieve.png", 100, 200)
        self.__mapa = Mapa(mazmorra, tesoro, textura)
        return self.__mapa

File: 3-Patron/Java/CreadorMapaJungla.java
package CreadoresDeMapa;
import buildercomposite.CreadorDeMapa;
import buildercomposite.IMapa;

import buildercomposite.Mapa;
import buildercomposite.Mazmorra;
import buildercomposite.Tesoro;
import buildercomposite.Textura;

public class CreadorMapaJungla extends CreadorDeMapa {

    public CreadorMapaJungla(IMapa mapa) {
        super(mapa);
    }

    @Override
    public Mapa CrearMapa() {
        Mazmorra mazmorra = new Mazmorra();
        Textura textura = new Textura("jugla.png",140,300);
        Tesoro tesoro = new Tesoro(100,2000);
        Mapa map = new Mapa(mazmorra,tesoro,textura);
        return map;
    }

    @Override
    public IMapa getMapa() {
        return super.getMapa(); //To change body of generated methods, choose Tools | Templates.
    }

    @Override
    public void setMapa(IMapa mapa) {
        this.mapa = mapa;
    }
    
     
    
   
    
}
